src/bptl/work_units/xential/client.py

```diff
@@ -60,11 +61,12 @@ def get_client(task: BaseTask, alias: str) -> "JSONClient":
     return _get_client(task, service, cls=client_classes.get(alias))
 
 
-def get_default_clients(alias: str) -> List["JSONClient"]:
-    # get the service and default credentials
-    # we can't use zgw_consumers client, since it uses OAS
-    clients = [
-        JSONClient(service, service.build_client().auth_header)
-        for service in Service.objects.filter(defaultservice__alias=alias).distinct()
-    ]
-    return clients
+def get_xential_client() -> "XentialClient":
+    default_service = DefaultService.objects.get(alias=XENTIAL_ALIAS)
+    service_credentials = AppServiceCredentials.objects.get(
+        service=default_service.service
+    )
```

Comment:
There is no guarantee that this object exists - it's possible that the credentials are defined on the `service` only.

Additionally, there's also no guarantee there is only one such object, as different apps may use different credentails for the same service, see the `unique_together` contraint in https://github.com/GemeenteUtrecht/bptl/blob/master/src/bptl/credentials/models.py#L35

`DefaultService` also has no unique constraint on the `alias` field, so you will get `MultipleObjectsReturned`, see https://github.com/GemeenteUtrecht/bptl/blob/master/src/bptl/tasks/models.py#L59

In short - the previous approach where you iterate over all services that have an alias to Xential can all yield valid clients. Probably the approach should be:

* determine all valid services based on the alias
* build clients for each service
* reduce the set of clients to unique clients (i.e. - only keep clients that differ in base endpoint/credentials)